### PR TITLE
Fix reports of operator-assignment, no-useless-escape, const-reassign

### PR DESCRIPTION
**Pull request by Josh-Cena**
#### Metadata

- [ ] Adds a new document
- [ ] Rewrites (or significantly expands) a document
- [x] Fixes a typo, bug, or other error

### CODE REVIEW

File: files/en-us/web/api/beforeunloadevent/index.md [hunk 1]
```diff
@@ -40,7 +40,7 @@ When a non-empty string is assigned to the `returnValue` Event property, a dialo
 
 ```js
 window.addEventListener("beforeunload", (event) => {
-  event.returnValue = "\o/";
+  event.returnValue = "o/";
```
Comment (teoli2003) [inline files/en-us/web/api/beforeunloadevent/index.md:43]: This is not an escape: this is a small human raising their arms: \o/

File: files/en-us/web/api/beforeunloadevent/index.md [hunk 2]
```diff
@@ -53,7 +53,7 @@ WebKit-derived browsers don't follow the spec for the dialog box. An almost-cros
 
 ```js
 window.addEventListener("beforeunload", (e) => {
-  const confirmationMessage = "\o/";
+  const confirmationMessage = "o/";
```
Comment (teoli2003) [inline files/en-us/web/api/beforeunloadevent/index.md:56]: This is not an escape: this is a small human raising their arms: \o/

File: files/en-us/web/javascript/guide/indexed_collections/index.md [hunk 3]
```diff
@@ -202,7 +202,7 @@ The function passed to `forEach` is executed once for every item in the array, w
 Note that the elements of an array that are omitted when the array is defined are not listed when iterating by `forEach`, but _are_ listed when `undefined` has been manually assigned to the element:
 
 ```js
-const array = ['first', 'second', , 'fourth'];
+const sparseArray = ['first', 'second', , 'fourth'];
 
 array.forEach((element) => {
```
Comment (teoli2003) [inline files/en-us/web/javascript/guide/indexed_collections/index.md:207]:
```suggestion
sparseArray.forEach((element) => {
```

File: files/en-us/web/javascript/guide/indexed_collections/index.md [hunk 4]
```diff
@@ -215,9 +215,9 @@ if (array[2] === undefined) {
   console.log('array[2] is undefined');  // true
```
Comment (teoli2003) [inline files/en-us/web/javascript/guide/indexed_collections/index.md:215]: Also one line above (214)

```suggestion
  console.log('sparseArray[2] is undefined');  // true
```

File: files/en-us/web/javascript/reference/functions/index.md [hunk 5]
```diff
@@ -472,11 +472,11 @@ still faster than `new Function(...)`. Therefore the `Function`
 constructor should generally be avoided whenever possible.
 
 It should be noted, however, that function expressions and function declarations nested
-within the function generated by parsing a `Function constructor` 's string
+within the function generated by parsing a `Function()` constructor's string
 aren't parsed repeatedly. For example:
 
 ```js
-const foo = (new Function("var bar = \'FOO!\';\nreturn(function() {\n\talert(bar);\n});"))();
+const foo = (new Function("var bar = 'FOO!'; return function() { alert(bar); };"))();
```
Comment (teoli2003) [inline files/en-us/web/javascript/reference/functions/index.md:479]:
```suggestion
const foo = (new Function("const bar = 'FOO!'; return function() { alert(bar); };"))();
```

File: files/en-us/web/javascript/reference/global_objects/object/defineproperty/index.md [hunk 6]
```diff
@@ -175,7 +175,7 @@ Object.defineProperty(o, 'a', {
 
 // Example of an object property added
 // with defineProperty with an accessor property descriptor
-const bValue = 38;
+let bValue = 38;
 Object.defineProperty(o, 'b', {
   get() { return bValue; },
   set(newValue) { bValue = newValue; },
```
Comment (teoli2003) [inline files/en-us/web/javascript/reference/global_objects/object/defineproperty/index.md:181]:
```suggestion
let value = 38;
Object.defineProperty(o, 'b', {
  get() { return value; },
  set(newValue) { value = newValue; },
```

**Review by teoli2003 — CHANGES_REQUESTED** [review]

File: files/en-us/web/api/beforeunloadevent/index.md (hunk 1 above)
Comment (Josh-Cena) [reply files/en-us/web/api/beforeunloadevent/index.md:43]: Haha, I get it now. Need an extra `\` in that case

File: files/en-us/web/javascript/reference/global_objects/object/defineproperty/index.md (hunk 6 above)
Comment (Josh-Cena) [reply files/en-us/web/javascript/reference/global_objects/object/defineproperty/index.md:181]: The `bValue` means "the value of `b`", not the type.

**Review by teoli2003 — APPROVED** [review]
👍

**Pull request merged**